### ISSUE TITLE
RF01 not always excluded in BigQuery

### ISSUE THREAD

**Issue by annebelleo** [sqlfluff/sqlfluff#7183]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Similar issues: #6682, #6521

In the SQLFluff [docs](https://docs.sqlfluff.com/en/stable/reference/rules.html#references-bundle), it states that the rule RF01 is excluded by default for some dialects, including BigQuery. However, there are some cases where an RF01 error still appears.

### Expected Behaviour

When the dialect is set to 'bigquery' in `.sqlfluff`, I would not expect to see an RF01 error.

### Observed Behaviour

This query does not give any errors:
```SQL
SELECT
    column_1,
    column_2
FROM foo,
    UNNEST(bar)

```

However, adding an alias does give an RF01 error:
```SQL
SELECT
    column_1,
    column_2
FROM foo,
    UNNEST(bar) AS foo_bar

```

```
L:   2 | P:   5 | RF01 | Reference 'column_1' refers to table/view not found in
                       | the FROM clause or found in ancestor statement.
                       | [references.from]
L:   3 | P:   5 | RF01 | Reference 'column_2' refers to table/view not found in
                       | the FROM clause or found in ancestor statement.
                       | [references.from]
L:   5 | P:  12 | RF01 | Reference 'bar' refers to table/view not found in the
                       | FROM clause or found in ancestor statement.
                       | [references.from]
```

Both are valid BigQuery queries when replacing "foo" with an existing table, the column names with existing columns, and "bar" with a column that can be unnested.

### How to reproduce

`test.sql`:
```SQL
SELECT
    column_1,
    column_2
FROM foo,
    UNNEST(bar) AS foo_bar

```

### Dialect

BigQuery

### Version

3.4.2

### Configuration

`.sqlfluff`:
```
[sqlfluff]
dialect = bigquery
```

`.python-version`: 
```
3.11
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
